src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -604,6 +623,7 @@ public void validateSQLSyntax(FailureCollector failureCollector, BigQuery bigQue
       try {
         bigQuery.create(JobInfo.of(queryJobConfiguration));
       } catch (BigQueryException e) {
+          LOG.error(String.format("Validation for SQL command %s failed.", sql), e);
```

Comment:
We should not log if we are throwing, it is anti-pattern.

Instead the error code as well while adding the error message below in failure collector.